web/src/components/molecules/Schema/FieldModal/FieldCreationModal/index.tsx

```diff
@@ -0,0 +1,245 @@
+import FieldValidationProps from "@reearth-cms/components/molecules/Schema/FieldModal/FieldValidationProps";
+import { SchemaFieldTypePropertyInput, SchemaFiledType } from "@reearth-cms/gql/graphql-client-api";
+
+import { fieldTypes } from "../../fieldTypes";
+import { FieldType } from "../../types";
+
+export interface FormValues {
+  title: string;
+  description: string;
+  key: string;
+  multiValue: boolean;
+  unique: boolean;
+  required: boolean;
+  type: SchemaFiledType;
+  typeProperty: SchemaFieldTypePropertyInput;
+}
+
+export interface Props {
+  open?: boolean;
+  selectedType: FieldType | null;
+  onClose?: (refetch?: boolean) => void;
+  onSubmit?: (values: FormValues) => Promise<void> | void;
+}
+
+const initialValues: FormValues = {
+  title: "",
+  description: "",
+  key: "",
+  multiValue: false,
+  unique: false,
+  required: false,
+  type: SchemaFiledType["Text"],
+  typeProperty: { text: { defaultValue: "", maxLength: 0 } },
+};
+
+const FieldCreationModal: React.FC<Props> = ({ open, onClose, onSubmit, selectedType }) => {
+  const [form] = Form.useForm();
+  const { TabPane } = Tabs;
+
+  const handleSubmit = useCallback(() => {
+    form
+      .validateFields()
+      .then(async values => {
+        values.type = selectedType;
+        if (selectedType === "Text") {
+          values.typeProperty = {
+            text: { defaultValue: values.defaultValue, maxLength: values.maxLength },
+          };
+        }
+        if (selectedType === "TextArea") {
+          values.typeProperty = {
+            textArea: { defaultValue: values.defaultValue, maxLength: values.maxLength },
+          };
+        }
+        if (selectedType === "MarkdownText") {
+          values.typeProperty = {
+            markdownText: { defaultValue: values.defaultValue, maxLength: values.maxLength },
+          };
+        }
+        if (selectedType === "Asset") {
+          values.typeProperty = {
+            asset: { defaultValue: values.defaultValue.uid },
+          };
+        }
+        if (selectedType === "Select") {
+          values.typeProperty = {
+            select: { defaultValue: values.defaultValue, values: values.values },
+          };
+        }
+        if (selectedType === "Integer") {
+          values.typeProperty = {
+            integer: { defaultValue: +values.defaultValue, min: +values.min, max: +values.max },
+          };
+        }
+        if (selectedType === "URL") {
+          values.typeProperty = {
+            url: { defaultValue: values.defaultValue },
+          };
+        }
+
+        await onSubmit?.(values);
+        form.resetFields();
+        onClose?.(true);
+      })
+      .catch(info => {
+        console.log("Validate Failed:", info);
+      });
+  }, [form, onClose, onSubmit, selectedType]);
+
+  const handleClose = useCallback(() => {
+    form.resetFields();
+    onClose?.(true);
+  }, [onClose, form]);
+
+  let additionalInput = <></>;
+  if (selectedType === "Select") {
```

Comment:
I think its better to just make the additionalInput a component and then inside the return do
```
{selectedType === "Select" && (<AdditionalInput />}
```